.github/workflows/cuda.yml

```
@@ -27,14 +27,32 @@ jobs:
             compiler: gcc
             python_version: "3.8"
             cuda_version: "11.5.1"
+            tree_learner: cuda
           - method: pip
             compiler: clang
             python_version: "3.9"
             cuda_version: "10.0"
+            tree_learner: cuda
           - method: wheel
             compiler: gcc
             python_version: "3.10"
             cuda_version: "9.0"
+            tree_learner: cuda
+          - method: source
+            compiler: clang
```

Thanks a lot for checking clang compiler for additional compilation warnings! I don't see any warnings. Please now revert gcc back.
```suggestion
            compiler: gcc
```